primitives/parachains/src/lib.rs

```diff
@@ -75,7 +75,7 @@ pub fn imported_parachain_head_storage_key_at_target(
 	para_id: ParaId,
 	head_hash: ParaHash,
 ) -> StorageKey {
-	bp_runtime::storage_double_map_final_key::<Blake2_128Concat, Twox64Concat>(
+	bp_runtime::storage_double_map_final_key::<Blake2_128Concat, Blake2_128Concat>(
```

Comment:
I'm not very familiar with storage values, but considering these fixes, I was wondering if we could use a safer approach here. I'm thinking of the following issues:
1. As seen here, we can use different hashers for the storage key vs the `StorageDoubleMap` that is defined in the pallet. This seems to be covered by unit tests however.
2. We can use different types when we decode the value. For example here the correct ones would be `ParaId`, `ParaHash`, but we might by mistake use something else when decoding. Not sure if this is covered by unit tests.

So I was wondering if we could use a safer approach to cover these issues. Maybe define a new structure to hold all these hashers and types and use if both when defining the `StorageDoubleMap` and when computing the key and reading the value. A scaled-down version of `StorageDoubleMap` maybe.

Reply:
Yep, that's a good idea. I'm not sure, though that you'll able to do that - all storage delarations are handled by procedural macro and it may happen that you can't specify e,g, `MyMap::Hasher` there. But you can try to experiment of course :)